components/HackathonModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-modal';
+import Image from '../components/Image';
+import Link from 'next/link';
+import styles from '../styles/components/HackathonModal.module.scss';
+import { website_config } from '../config';
+
+const MODAL_DISPLAY_INTERVAL = 2 * 24 * 60 * 60 * 1000; // 2 days in milliseconds
+
+interface HackathonModalProps {
+  onRequestClose: () => void;
+}
+
+const HackathonModal: React.FC<HackathonModalProps> = ({ onRequestClose }) => {
```

Comment:
```suggestion
const HackathonModal = ({ onRequestClose }: HackathonModalProps) => {
```
The `React.FC` is just extra gibrish here it adds no type safety really or benefit lets drop it and just type the props. ts will infer the rest.